types/rest/rest.go

```diff
@@ -428,6 +428,38 @@ func GetRequest(url string) ([]byte, error) {
 	return body, nil
 }
 
+// GetRequestWithHeaders defines a wrapper around an HTTP GET request with a provided URL
+// and custom headers
+// An error is returned if the request or reading the body fails.
+func GetRequestWithHeaders(url string, headers map[string]string) ([]byte, error) {
```

Comment:
This function is only used in one test file, right? I would prefer to put all test-only functions in `_test.go` files, or maybe in testutil.

At first glance I thought this was used somewhere in the non-test code.